install/tests/Makefile

```diff
@@ -37,286 +21,317 @@ help: Makefile
 get-kubeconfig:
-	echo "Getting kubeconfig for $$TF_VAR_TEST_ID terraform state" && \
+	@echo "Getting kubeconfig for $$TF_VAR_TEST_ID terraform state" && \
     export provider=$$(echo "$$TF_VAR_TEST_ID" | sed 's/\(.*\)-/\1 /' | xargs | awk '{print $$2}') && \
+	export provider=k3s && \
 	$(MAKE) $$provider-kubeconfig && echo "kubeconfig written to ${KUBECONFIG}"
 
+sync-kubeconfig:
+	[[ -z $$self_hosted_jobs ]] || gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS} --project=sh-automated-tests
+	gcloud config set project sh-automated-tests
+	gsutil cp gs://nightly-tests/tf-state/${TF_VAR_TEST_ID}-kubeconfig ${KUBECONFIG} || echo "No kubeconfig"
+
+## k3s-kubeconfig: Get the kubeconfig configuration for GCP K3s
+k3s-kubeconfig: sync-kubeconfig
+
+## gcp-kubeconfig: Get the kubeconfig configuration for GCP GKE
+gcp-kubeconfig:
+	@[[ -n $$self_hosted_jobs ]] || $(MAKE) download-creds
+	@[[ -n $$self_hosted_jobs ]] || cat $(TF_VAR_TEST_ID)-output.json | yq r - 'gke_user_key.value' | tr -d '\n' | jq -Rsa . > ${TF_VAR_TEST_ID}-key.json
+	@[[ -f ${TF_VAR_TEST_ID}-key.json ]] || cat ${GOOGLE_APPLICATION_CREDENTIALS} > ${TF_VAR_TEST_ID}-key.json
+	@gcloud auth activate-service-account --key-file=${TF_VAR_TEST_ID}-key.json --project=sh-automated-tests || { echo "Count not authenicate the service account"; exit 1; }
+	@export KUBECONFIG=${KUBECONFIG} && \
+	gcloud container clusters get-credentials gp-${TF_VAR_TEST_ID} --zone europe-west1-d --project sh-automated-tests || echo "No cluster present"
+	@rm -f ${TF_VAR_TEST_ID}-key.json
+	@echo "Kubeconfig written to ${KUBECONFIG}"
+
+## azure-kubeconfig: Get the kubeconfig configuration for Azure AKS
+azure-kubeconfig:
+	@[[ -z "$$self_hosted_jobs" ]] || az login --service-principal -u $$ARM_CLIENT_ID -p $$ARM_CLIENT_SECRET  --tenant $$ARM_TENANT_ID
+	@export KUBECONFIG=${KUBECONFIG} && \
+	az aks get-credentials --name gp-$$TF_VAR_TEST_ID-cluster --resource-group gp-$$TF_VAR_TEST_ID --file ${KUBECONFIG} || echo "No cluster present"
+
+get-aws-secret:
+	@gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS} --project=sh-automated-tests
+	@gcloud config set project sh-automated-tests
+	@gsutil cp gs://nightly-tests/tf-state/${TF_VAR_TEST_ID}-creds.json ${TF_VAR_TEST_ID}-output.json
+	@echo "export AWS_SECRET_ACCESS_KEY=$$(cat ${TF_VAR_TEST_ID}-output.json | yq r - 'aws_cluster_user.value.secret_access_key')" > ${TF_VAR_TEST_ID}-creds
+	@echo "export AWS_ACCESS_KEY_ID=$$(cat ${TF_VAR_TEST_ID}-output.json | yq r - 'aws_cluster_user.value.access_key_id')" >>  ${TF_VAR_TEST_ID}-creds
+	@echo -e "\033[0;33mAWS service account credentials fetched, run 'source $$(pwd)/$${TF_VAR_TEST_ID}-creds' to load them into your environment\033[0;m"
+
+## aws-kubeconfig: Get the kubeconfig configuration for AWS EKS
+aws-kubeconfig:
+	@[[ -n $$self_hosted_jobs ]] || $(MAKE) get-aws-secret
+	@[[ -f ${TF_VAR_TEST_ID}-creds ]] || touch ${TF_VAR_TEST_ID}-creds
```

Comment:
**question:** Under which conditions would `make get-aws-secret` fail to generate a credentials file, but touching the credential file would let the rest of test execution work? I ran into this when chasing down some other failures and didn't understand what's going on; if there's some subtle behavior at play can we document what's going in via a comment?

Reply:
It would fail if there is no such credentials file present in GCS. We are not catching this error anymore. This would fail [here](https://github.com/gitpod-io/gitpod/pull/13542/files/021b9bf50683f76951eedf3f191f683a23b3a73e#diff-bd7b7f3a397df93111a6b6e2a4284b01bd57e04474bc9f91f1b02ee1960399a2R174) with the error